### DISCUSSION TITLE
classes/class_textedit

### DISCUSSION THREAD

**Discussion by giscus** [godotengine/godot-docs-user-notes#211] [User-contributed notes]
# classes/class_textedit

Inherits: Control< CanvasItem< Node< Object Inherited By: CodeEdit A multiline text editor. Description: A multiline text editor. It also has limited facilities for editing code, such as syntax hig...

https://docs.godotengine.org/en/latest/classes/class_textedit.html

**Comment by Dfred** [comment]
the documentation should mention when [is_caret_visible()](https://docs.godotengine.org/en/4.3/classes/class_textedit.html#class-textedit-method-is-caret-visible) returns `false`, as it will return `true` even if the caret is actually not visible, such as when `scroll_fit_content_height = true`, and the caret is beyond displayed lines

**Reply by Calinou** [reply]
Pull request opened:

- https://github.com/godotengine/godot/pull/98423

**Comment by CatzHoekk** [comment]
If you set set_tooltip_request_func but don't want to show a tooltip in certain situations just return ""

This is "somewhat" documented in Control::get_tooltip

**Reply by Calinou** [reply]
Pull request opened:

- https://github.com/godotengine/godot/pull/107980

**Comment by Jesaja-Schlotzer** [comment]
I tried to fill a TextEdit with the [set_line()](https://docs.godotengine.org/en/4.5/classes/class_textedit.html#class-textedit-method-set-line) method today. That didn't work as I expected so I looked at the source code and found that the method only sets a line if the line index is less than or equal to the current number of lines of the TextEdit. Is that behavior maybe relevant for the documentation? Even if the method probably is just not intended for that use case.

My workaround is to just add linebreaks till there are enought lines. But I would love to know if there is a better solution than this ^^'.
```gdscript
while line_index > text_edit.get_line_count():
	text_edit.text += "\n"
text_edit.text += line
```

**Reply by kitbdev** [reply]
Yes, `set_line` is for replacing the content of an existing line, not really adding new ones. If you want to add a line that is multiple lines after the end, you can also use [insert_text](https://docs.godotengine.org/en/latest/classes/class_textedit.html#class-textedit-method-insert-text), like this:
```gdscript
var last_line := text_edit.get_line_count() - 1
var last_line_length := text_edit.get_line(last_line).length()
var newlines_to_add := 3
text_edit.insert_text("\n".repeat(newlines_to_add) + line, last_line, last_line_length)
```
